### PR TITLE
feat: [CDS-87940]: Add Pipeline setting to Mandate pipeline create and edit permission to create and edit triggers

### PR DESCRIPTION
**Pull request by pankajsingh-harness**
### Summary

Add a new Pipeline Setting to Mandate pipeline create and edit permission to create and edit triggers.

Default value `MANDATE_PIPELINE_CREATE_EDIT_PERMISSION_TO_CREATE_EDIT_TRIGGERS: 'true'` 

Before: The user must have both `EXECUTE_PIPELINE` & `EDIT_PIPELINE` permission to create and edit triggers.

After: If `MANDATE_PIPELINE_CREATE_EDIT_PERMISSION_TO_CREATE_EDIT_TRIGGERS: false` user can create and edit the trigger only with `EXECUTE_PIPELINE` permission

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/72-triggers/components/Triggers/useIsTriggerCreatePermission.tsx [hunk 1]
```diff
@@ -11,10 +11,36 @@ import type { PipelinePathProps } from '@common/interfaces/RouteInterfaces'
...
+    pipelineIdentifier,
+    accountId: accountIdentifier,
+    orgIdentifier,
+    projectIdentifier
+  } = useParams<PipelinePathProps>()
   const [isTriggerCreatePermission, setIsTriggerCreatePermission] = useState(false)
+  const [
+    isMandatePipelineCreateEditPermissionCreateEditTriggers,
+    setIsMandatePipelineCreateEditPermissionCreateEditTriggers
+  ] = useState(true)
```
Comment (rameshx) [inline src/modules/72-triggers/components/Triggers/useIsTriggerCreatePermission.tsx:28]: this state is not needed as `isMandatePipelineCreateEditPermissionCreateEditTriggers` can be derived from `projectSettingData`

Comment (chetanvekariya-harness) [reply src/modules/72-triggers/components/Triggers/useIsTriggerCreatePermission.tsx:28]: +1

**Review by rameshx — APPROVED** [review]

**Review by sonaligoyal-harness — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks good

**Review by sonaligoyal-harness — APPROVED** [review]

File: src/modules/72-triggers/pages/triggers/TriggerLandingPage/__tests__/TriggerDetailPage.test.tsx [hunk 2]
```diff
@@ -98,9 +104,85 @@ describe('Test Trigger Detail Page Test', () => {
...
+      jest.spyOn(cdNg, 'useGetSettingValue').mockImplementation((): any => {
+        return { data: { data: { value: 'true' } } }
+      })
+    })
+
+    test('EXECUTE_PIPELINE: false, EDIT_PIPELINE: true', () => {
+      jest.spyOn(usePermission, 'usePermission').mockImplementation(() => [false, true])
+
+      const { container } = render(<TestComponent />)
+      expect(container.querySelector('[aria-label="edit"]')).toBeDisabled()
```
Comment (rameshx) [inline src/modules/72-triggers/pages/triggers/TriggerLandingPage/__tests__/TriggerDetailPage.test.tsx:128]: `getByLabelText` can be used for aria labels

**Review by rameshx — APPROVED** [review]

**Review by ishantmehtaharness — APPROVED** [review]
Looks good. Thanks for adding test cases

File: src/modules/72-triggers/pages/triggers/TriggerLandingPage/__tests__/TriggerDetailPage.test.tsx [hunk 3]
```diff
@@ -70,6 +72,10 @@ jest.mock('@harnessio/react-pipeline-service-client', () => ({
 
 describe('Test Trigger Detail Page Test', () => {
   test('should test snapshot view', async () => {
+    jest.spyOn(usePermission, 'usePermission').mockImplementation(() => [true, true])
```
Comment (chetanvekariya-harness) [inline src/modules/72-triggers/pages/triggers/TriggerLandingPage/__tests__/TriggerDetailPage.test.tsx:75]: This might not be needed.

**Pull request merged**